config/samples/gcp/gcp_openshift_route.yaml

```diff
@@ -0,0 +1,22 @@
+apiVersion: externaldns.olm.openshift.io/v1alpha1
+kind: ExternalDNS
+metadata:
+  name: sample-gcp
+spec:
+  domains:
+  - filterType: Include
+    matchType: Exact
+    name: test-gcp1.qe.gcp.devcluster.openshift.com
+  provider:
+    type: GCP
+  source:
+    # Incase you have multiple ingress controllers you must specify ingress controller name in the routerName
+    # so that the external dns will use the router canonical name correrponding to it to create a dns record. 
+    openshiftRouteOptions:  
+      routerName: default
+    #Source Type is route resource of OpenShift    
+    type: OpenShiftRoute
+    fqdnTemplate: 
+    - "{{.Name}}.apps.test-gcp1.qe.gcp.devcluster.openshift.com"
```

Comment:
Can this ever work in the domain filter is the exact match of `test-gcp1.qe.gcp.devcluster.openshift.com` ? I believe that `app` subdpmaon should be omitted.

Reply:
Not sure what you mean. I have used CR from real env. test-gcp1.qe.gcp.devcluster.openshift.com  Exact means apps.test-gcp1.qe.gcp.devcluster.openshift.com also works. You can prepend that with apps or anything. Putting apps is important as the ingress controller creates a wildcard for *.apps .

Reply:
Also in real case fqdnTemplate:  is useless as hostname will always be available in the route as we have a check openshift api server for it's existence.  I had to add it as webhook does not allow HostNameAnnotation Ignore with no fqdnTemplate.

Reply:
Yes, you are right, `FQDNTemplate` not really needed for the OpenShift case. Let's remove it then to showcase this.

Reply:
Yes, you are right, the validation in the operator mirrors the same validation in external-dns: no matter the source - it's not possible to not have `FQDNTemplate` if the hostname annotatin policy is ignore.